src/Features/CSharp/Portable/GoToDefinition/CSharpGoToDefinitionSymbolService.cs

```diff
@@ -20,5 +21,90 @@ public CSharpGoToDefinitionSymbolService()
 
         protected override ISymbol FindRelatedExplicitlyDeclaredSymbol(ISymbol symbol, Compilation compilation)
             => symbol;
+
+        protected override int? GetTargetPositionIfControlFlow(SyntaxToken token)
+        {
+            var node = token.Parent;
+            if (node is null)
+            {
+                return null;
+            }
+
+            switch (token.Kind())
+            {
+                case SyntaxKind.ContinueKeyword:
+                    var foundContinuedLoop = TryFindContainingLoop(node);
+
+                    return foundContinuedLoop?.IsContinuableConstruct() == true
+                        ? foundContinuedLoop.GetFirstToken().Span.Start
+                        : null;
+
+                case SyntaxKind.BreakKeyword:
+                    var foundBrokenLoop = TryFindContainingLoop(node);
+
+                    return foundBrokenLoop?.IsBreakableConstruct() == true
+                        ? foundBrokenLoop.GetLastToken().Span.End
+                        : null;
+
+                case SyntaxKind.YieldKeyword:
+                    var nextToken = token.GetNextToken();
+                    switch (nextToken.Kind())
+                    {
+                        case SyntaxKind.ReturnKeyword:
+                            return GetTargetPositionIfControlFlow(nextToken);
+
+                        case SyntaxKind.BreakKeyword:
+                            var foundReturnableConstruct = TryFindContainingReturnableConstruct(node);
+                            return foundReturnableConstruct?.GetLastToken().Span.End;
```

Comment:
not srue how i feel about this.  I feel that `yield break;` doesn't really go anywhere.  going to the end brace of amethod just doesn't feel helpful.  Like... hwat would i do at that point? Do you feel differently?

Reply:
The choice for `yield break` is out of consistency with `return` in my opinion. But I agree that it's less useful, since we don't have iterator lambdas.

Reply:
Note: we may get iterator lambdas if we move forward with allowing explicit return type of lambda.

Reply:
@CyrusNajmabadi Sam said he's okay with using the end of the method as target for `yield break`. Should I also check with Jason or some other folks?

Reply:
yeah, i would check with more people on this.  end-of-method seems super weird.  it's not a thing that we generally do/indicate in any other way.